app/controllers/queue_items_controller.rb

```diff
@@ -10,33 +10,66 @@ def create
     queue_the_item(video)
     redirect_to my_queue_path
   end
-  
+
   def destroy
     queued_item = QueueItem.find(params[:id])
-    queued_item.destroy if belongs_to_current_user?(queued_item)
- 
+    queued_item.destroy if current_user.owns_item?(queued_item)
+    normalize_queue_item_positions
+    
+    redirect_to my_queue_path
+  end
+  
+  # sort the queue items by their list position
+  def sort
+    redirect_to my_queue_path and return if invalid_inputs?
```

Comment:
Transactions basically does all the heavy lifting wrt rolling back invalid data for you but if you are dealing with it at this level and it works I din't think it's a big issue.